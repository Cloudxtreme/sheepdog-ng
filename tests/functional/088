#!/bin/bash

# Test data integrity after changing my address (sheep -y <myaddr>)

. ./common

MY_ADDR=`hostname -I | awk '{print $1}'`
ORIG_SHEEP_OPTIONS=$SHEEP_OPTIONS
NEW_SHEEP_OPTIONS="-n -y $MY_ADDR -l level=debug"

echo "1. start sheep with my address"
SHEEP_OPTIONS=$NEW_SHEEP_OPTIONS
_start_sheep 0
_wait_for_sheep 1
_cluster_format -c 1
$DOG cluster info | _filter_cluster_info | _filter_ip
$DOG vdi create test 40M -P
_vdi_list
$DOG cluster shutdown
_wait_for_sheep_stop

echo "2. start sheep with default"
SHEEP_OPTIONS=$ORIG_SHEEP_OPTIONS
_start_sheep 0
_wait_for_sheep 1
$DOG cluster info | _filter_cluster_info | _filter_ip
$DOG node info
$DOG cluster shutdown
_wait_for_sheep_stop

echo "3. start sheep with my address again"
SHEEP_OPTIONS=$NEW_SHEEP_OPTIONS
_start_sheep 0
_wait_for_sheep 1
$DOG cluster info | _filter_cluster_info | _filter_ip
_vdi_list

_start_sheep 1
_wait_for_sheep 2
$DOG vdi check test
_node_info
