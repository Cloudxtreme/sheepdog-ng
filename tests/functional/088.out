QA output created by 088
1. start sheep with my address
using backend plain store
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version
DATE      1 [IP:7000]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         0   40 MB   40 MB  0.0 MB DATE   7c2b25      1              
2. start sheep with default
Cluster status: Waiting for other nodes to join cluster

Cluster created at DATE

Epoch Time           Version
DATE      1 [IP:7000]
Response's result: Waiting for other nodes to join cluster
Cannot get information from any nodes
Id	Size	Used	Avail	Use%
3. start sheep with my address again
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version
DATE      1 [IP:7000]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         0   40 MB   40 MB  0.0 MB DATE   7c2b25      1              
finish check&repair test
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	5	0	0	0	0	0
1	5	1	0	0	0	0
